### DISCUSSION TITLE
traverse the entire Rtree to get the index size

### DISCUSSION THREAD

**Discussion by JunoWang** [boostorg/geometry#970] [Q&A] [answered]
I am looking for the boost tree's size after bulk loading my data into boost Rtree, and I found it is hard to traverse the whole tree to get the sizes I am looking for. So is there any possibility that can get the entire R-tree's size in bytes by traversing the whole tree ( all internal node and leaf node)?

**Comment by awulkiew** [comment, marked as answer]
I don't know what would you like to do exactly and what do you mean by "sizes". But here is the general answer:

The number of elements is returned by [`rtree.size()`](https://www.boost.org/doc/libs/1_78_0/libs/geometry/doc/html/geometry/reference/spatial_indexes/boost__geometry__index__rtree/size__.html)

The bounding box enclosing all of the elements is returned by [`rtree.bounds()`](https://www.boost.org/doc/libs/1_78_0/libs/geometry/doc/html/geometry/reference/spatial_indexes/boost__geometry__index__rtree/bounds__.html)

You can iterate through all of the elements stored in the rtree using [`rtree.begin()`](https://www.boost.org/doc/libs/1_78_0/libs/geometry/doc/html/geometry/reference/spatial_indexes/boost__geometry__index__rtree/begin__.html) and [`rtree.end()`](https://www.boost.org/doc/libs/1_78_0/libs/geometry/doc/html/geometry/reference/spatial_indexes/boost__geometry__index__rtree/end__.html) or just using the rtree in range for loop.

The numbers of internal nodes, leafs and other statistics can be gathered using this utility:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/index/detail/rtree/utilities/statistics.hpp#L91

If you need something else you have to write your own visitor using the `view` used by the utilities implemented here:
https://github.com/boostorg/geometry/tree/develop/include/boost/geometry/index/detail/rtree/utilities
e.g. like it is done in statistics utility:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/index/detail/rtree/utilities/statistics.hpp#L26-L103
or by the one checking correctness of bounding boxes of nodes:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/index/detail/rtree/utilities/are_boxes_ok.hpp#L26-L131

**Reply by JunoWang** [reply]
I mean the size is how many bytes the r-tree will take in memory, this will consider each node(internal node, and leaf node) how many bytes the pointer, MBR, and other variable stored in the nodes will take. 
thank you for replying!
